packages/data-stores/src/mutations/use-domains-bulk-actions-mutation.ts

```diff
@@ -49,12 +55,26 @@ export function useDomainsBulkActionsMutation<
 					} );
 			}
 		},
+		onSuccess: ( data, variables ) => {
+			if ( variables.type !== 'set-auto-renew' ) {
+				return;
+			}
+
+			// Makes sure the success notice is shown once statuses have been updated
+			queryClient.invalidateQueries( { queryKey: getBulkDomainUpdateStatusQueryKey() } );
+
+			// Refreshes data for all the domains that were updated
+			variables.blogIds.forEach( ( blogId ) => {
+				queryClient.invalidateQueries( { queryKey: getSiteDomainsQueryKey( blogId ) } );
+				queryClient.invalidateQueries( { queryKey: getAllDomainsQueryKey( { no_wpcom: true } ) } );
```

Comment:
In general the code looks good, but I have concern due to this line.
If we have a few `blogIds`, that meen that we do invalidation by `no_wpcom: true` a few times. I didn't dig deeper to understand the reason, but if the key is static, looks like we should run it only once, after the `forEach`. Or I am missing something?

Reply:
> In general the code looks good, but I have concern due to this line. If we have a few `blogIds`, that meen that we do invalidation by `no_wpcom: true` a few times. I didn't dig deeper to understand the reason, but if the key is static, looks like we should run it only once, after the `forEach`. Or I am missing something?

Ha, it looks this is the same line I flagged.